ReactiveCocoa/Swift/Signal.swift

```diff
@@ -273,26 +273,29 @@ public func combineLatestWith<T, U, E>(otherSignal: Signal<U, E>)(signal: Signal
 	return Signal { observer in
 		let lock = NSRecursiveLock()
 		lock.name = "org.reactivecocoa.ReactiveCocoa.combineLatestWith"
-
+		
 		let signalState = CombineLatestState<T>()
 		let otherState = CombineLatestState<U>()
-
+		
 		let onBothNext = { () -> () in
-			let combined = (signalState.latestValue!, otherState.latestValue!)
-			sendNext(observer, combined)
+			sendNext(observer, (signalState.latestValue!, otherState.latestValue!))
 		}
-
+		
 		let onError = { sendError(observer, $0) }
 		let onBothCompleted = { sendCompleted(observer) }
 		let onInterrupted = { sendInterrupted(observer) }
-
+		
 		let signalDisposable = signal |> observeWithStates(signalState, otherState, lock, onBothNext, onError, onBothCompleted, onInterrupted)
 		let otherDisposable = otherSignal |> observeWithStates(otherState, signalState, lock, onBothNext, onError, onBothCompleted, onInterrupted)
-
+		
 		return CompositeDisposable(ignoreNil([ signalDisposable, otherDisposable ]))
 	}
 }
 
+public func combineLatestWith<T, E>(otherSignal: Signal<T, E>)(signal: Signal<[T], E>) -> Signal<[T], E> {
```

Comment:
Oh, I made the equivalent of the signal producer private. Made this private too now :)